### PR TITLE
[batch] Add billing UI page

### PR DESCRIPTION
**Pull request by jigold**
I added a billing UI page. I thought about supporting search/filter on user and billing project, but it was too hard to get the formatting how I wanted and thought this was good enough for now.

@cseed Can you confirm you're okay with the query I added over batches? I also added an index on `time_created` on the batches table, so this will require a migration.

image

### CODE REVIEW

**Comment by jigold** [comment]
Also, maybe we should search on time_created for START and time_completed for END.

**Comment by johnc1231** [comment]
Looks good to me. Do you want me to wait to approve until Cotton looks at your SQL query?

**Comment by johnc1231** [comment]
@jigold Do you want me to approve this or still waiting on Cotton?

**Comment by jigold** [comment]
@cseed I have a VM ready that should be able to redeploy batch once this goes in. Can you confirm you're okay with this before John approves?

Also, FYI @danking

**Comment by danking** [comment]
bump @cseed. Reassigning to you since you're the blocker.

File: batch/batch/front_end/front_end.py [hunk 1]
```diff
@@ -1075,6 +1077,95 @@ def check_service_account_permissions(user, sa):
...
+    except ValueError:
+        return await parse_error(f"Invalid value for end '{end_query}'; must be in the format of MM/DD/YYYY.")
+
+    if start > end:
+        return await parse_error(f'Invalid search; start must be earlier than end.')
+
+    sql = f'''
+SELECT billing_project, user, CAST(COALESCE(SUM(msec_mcpu), 0) AS SIGNED) AS msec_mcpu
+FROM batches
+WHERE batches.`time_created` >= %s AND batches.`time_created` <= %s
```
Comment (cseed) [inline batch/batch/front_end/front_end.py]: This was the query you were worried about?  You've got the right index.  Looks good to me.  Ultimately we might want to limit the search range, but this seems great for now.

**Comment by danking** [comment]
erm this isn't merging.

**Comment by cseed** [comment]
Does your PR I need to review fix this? :-)

**Comment by jigold** [comment]
No, I have the WIP label on so it doesn't merge until I'm ready.

**Comment by danking** [comment]
Ah, so the open batches weren't breaking merge then

**Pull request merged**